widget/button.go

```diff
@@ -308,6 +333,8 @@ func (r *buttonRenderer) buttonColor() color.Color {
 	switch {
 	case r.button.Disabled():
 		return theme.DisabledButtonColor()
+	case r.button.focused:
+		return theme.FocusColor()
```

Comment:
I am unsure if this is the best way to indicate the button is focused 🤔

Reply:
Well, it matches how we have done it for menus - what alternative would be more suitable?

Reply:
I think material design specs are suitable for this case:
![image](https://user-images.githubusercontent.com/12239342/119898942-15b8ac80-bf08-11eb-8fb7-42e2be8fa487.png)

However as it was discussed before, those state colors are confusing among all the material design components. But definitely I don't think buttons should change their background color when they are focused (an overlay color is ok but a full color replacement does not seem great).

Another possibility is to have a focused border in buttons (like the hyperlinks), just like github buttons:

https://user-images.githubusercontent.com/12239342/119899651-0128e400-bf09-11eb-81ed-16188176b50a.mp4

Reply:
> Maybe the addition of SelectionColor and FocusColor as in Material Design could be enough?

This should be right way to go IMHO.

Reply:
> Maybe the addition of SelectionColor and FocusColor as in Material Design could be enough?
> But definitely I don't think buttons should change their background color when they are focused

This usage is consistent with Menu, Check, Radio and others. I am using `FocusColor` to indicate focus - I don't see how adding `SelectionColor` helps. We went with a not-grey colour because keyboard is an important part of accessibility and a really subtle indicator is not appropriate.

At this point I think that a request to change how Focus colour is handled should be a separate issue - I don't want to have to change it across all for this ticket and I also don't want to introduce inconsistencies with this PR.

Reply:
> This usage is consistent with Menu, Check, Radio and others.

Menu, check and radio don't have a configurable specific background through the theme iirc. But buttons do, so the focus color is going to change the whole appearance of a button with no transparent background (would look like a completely different one). This problem would be bigger if primary, secondary, success, danger, etc. colored variants for buttons are added.

> We went with a not-grey colour because keyboard is an important part of accessibility and a really subtle indicator is not appropriate.

That's fine, but it doesn't feel good when you have a widget with a custom background, IMO. Maybe a focused border color should be used for widgets that have background colors different from canvas color? 🤔

> At this point I think that a request to change how Focus colour is handled should be a separate issue - I don't want to have to change it across all for this ticket and I also don't want to introduce inconsistencies with this PR.

Oh sorry, that was not my intention (proposing the Focus color change in this PR), I was just exposing two possible solutions to avoid having to change the whole button background color when it is focused. The first one (changing the FocusColor) I agree should be done in a different PR (if it is decided) and the second one is having a focused border just like the hyperlink (that could be done in this PR). Personally, I think changing completely the background color of the widgets to represent a focus state does not feel natural. 

Maybe this is a subjective area and more opinions are needed?

Reply:
Ah yes, it should be blended - like the hover colour - sorry I missed that.
Should be fixed now.